### ISSUE TITLE
@Vector(2, u4) generates llvm.getelementptr that produces wrong code

### ISSUE THREAD

**Issue by xxxbxxx** [ziglang/zig#11998]
### Zig Version

0.10.0-dev.2868+dcd8d3728

### Steps to Reproduce

```zig
pub fn main() anyerror!void {
    const Vec2 = @Vector(2, u4);
    var x: u4 = 1;
    const val = Vec2{ x, 3 };
    if (val[1] == 0) unreachable;
}
```

### Expected Behavior

runs.  
(and it does with stage1)

### Actual Behavior

with stage2, generated executable crashes.

**Comment by xxxbxxx** [comment]
The generated llvm ir for stage1 is:

```llvm
; Function Attrs: nobuiltin nounwind
define internal fastcc i16 @main(%std.builtin.StackTrace* nonnull %0) unnamed_addr #1 !dbg !15027 {
Entry:
  %result = alloca i16, align 2
  %x = alloca i4, align 1
  %val = alloca <2 x i4>, align 1
  store i4 1, i4* %x, align 1, !dbg !15036
  call void @llvm.dbg.declare(metadata i4* %x, metadata !15032, metadata !DIExpression()), !dbg !15036
  %1 = load i4, i4* %x, align 1, !dbg !15037
  %2 = load <2 x i4>, <2 x i4>* %val, align 1, !dbg !15037
  %3 = insertelement <2 x i4> %2, i4 %1, i32 0, !dbg !15037
  store <2 x i4> %3, <2 x i4>* %val, align 1, !dbg !15037
  %4 = load <2 x i4>, <2 x i4>* %val, align 1, !dbg !15038
  %5 = insertelement <2 x i4> %4, i4 3, i32 1, !dbg !15038
  store <2 x i4> %5, <2 x i4>* %val, align 1, !dbg !15038
  call void @llvm.dbg.declare(metadata <2 x i4>* %val, metadata !15034, metadata !DIExpression()), !dbg !15039
  %6 = load <2 x i4>, <2 x i4>* %val, align 1, !dbg !15040
  %7 = extractelement <2 x i4> %6, i32 1, !dbg !15040
  %8 = icmp eq i4 %7, 0, !dbg !15041
  br i1 %8, label %Then, label %Else, !dbg !15042

Then:                                             ; preds = %Entry
  call fastcc void @std.builtin.default_panic(%"[]u8"* @93, %std.builtin.StackTrace* null), !dbg !15043
  unreachable, !dbg !15043

Else:                                             ; preds = %Entry
  br label %EndIf, !dbg !15042

EndIf:                                            ; preds = %Else
  store i16 0, i16* %result, align 2, !dbg !15044
  ret i16 0, !dbg !15045
}
```


for stage 2:
```llvm
; Function Attrs: nounwind
define internal fastcc i16 @vecinit.main(%builtin.StackTrace* nonnull %0) unnamed_addr #0 !dbg !3811 {
Entry:
  %1 = alloca <2 x i4>, align 1
  %2 = alloca i4, align 1
  store i4 1, i4* %2, align 1, !dbg !3821
  call void @llvm.dbg.declare(metadata i4* %2, metadata !3816, metadata !DIExpression()), !dbg !3821
  %3 = getelementptr inbounds <2 x i4>, <2 x i4>* %1, i32 0, i64 0, !dbg !3822
  %4 = load i4, i4* %2, align 1, !dbg !3822
  store i4 %4, i4* %3, align 1, !dbg !3822
  %5 = getelementptr inbounds <2 x i4>, <2 x i4>* %1, i32 0, i64 1, !dbg !3822
  store i4 3, i4* %5, align 1, !dbg !3822
  call void @llvm.dbg.declare(metadata <2 x i4>* %1, metadata !3819, metadata !DIExpression()), !dbg !3822
  %6 = load <2 x i4>, <2 x i4>* %1, align 1, !dbg !3823
  %7 = extractelement <2 x i4> %6, i64 1, !dbg !3823
  %8 = icmp eq i4 %7, 0, !dbg !3823
  br i1 %8, label %Then, label %Else, !dbg !3823

Then:                                             ; preds = %Entry
  call fastcc void @builtin.default_panic(i8* getelementptr inbounds ([24 x i8], [24 x i8]* @vecinit.main__anon_3602, i32 0, i32 0), i64 24, %builtin.StackTrace* null), !dbg !3823
  unreachable, !dbg !3823

Else:                                             ; preds = %Entry
  br label %Block, !dbg !3823

Block:                                            ; preds = %Else
  ret i16 0, !dbg !3823
}
```

**Comment by xxxbxxx** [comment]
corresponding AIR:

```
  %18 = alloc(*const @Vector(2, u4))
  %22 = bitcast(*@Vector(2, u4), %18)
  %25 = ptr_elem_ptr(%22, %23!)
  %27 = load(u4, %9!)
  %29!= store(%25!, %27!)
  %32 = ptr_elem_ptr(%22!, %30!)
  %37!= store(%32!, %35!)
```

**Comment by Vexu** [comment]
Fixed by #13959
